tests/UnitTests/Benchmark/TextBenchmarkTests.mm

```diff
@@ -0,0 +1,536 @@
+public:
+    TextBenchmarkBase() {
+        auto colorspace = woc::MakeAutoCF<CGColorSpaceRef>(CGColorSpaceCreateDeviceRGB());
+        m_context = woc::MakeAutoCF<CGContextRef>(CGBitmapContextCreate(nullptr,
+                                                                        sc_defaultSize.width,
+                                                                        sc_defaultSize.height,
+                                                                        8,
+                                                                        sc_defaultSize.width * 4,
+                                                                        colorspace,
+                                                                        kCGImageAlphaPremultipliedFirst));
+    }
+
+protected:
+    woc::AutoCF<CGContextRef> m_context;
+};
+
+class CTFrameDrawBase : public TextBenchmarkBase {
+public:
+    CTFrameDrawBase() {
+        CTParagraphStyleSetting setting;
+        CTTextAlignment alignment = kCTCenterTextAlignment;
+        setting.spec = kCTParagraphStyleSpecifierAlignment;
+        setting.valueSize = sizeof(CTTextAlignment);
+        setting.value = &alignment;
+        auto paragraphStyle = woc::MakeAutoCF<CTParagraphStyleRef>(CTParagraphStyleCreate(&setting, std::extent<decltype(setting)>::value));
+        UIFont* font = [UIFont systemFontOfSize:24];
+
+        CFStringRef keys[2] = { kCTFontAttributeName, kCTParagraphStyleAttributeName };
+        CFTypeRef values[2] = { font, paragraphStyle };
+        auto dict = woc::MakeAutoCF<CFDictionaryRef>(CFDictionaryCreate(nullptr,
+                                                                        (const void**)keys,
+                                                                        (const void**)values,
+                                                                        std::extent<decltype(keys)>::value,
+                                                                        &kCFTypeDictionaryKeyCallBacks,
+                                                                        &kCFTypeDictionaryValueCallBacks));
+
+        auto attrString = woc::MakeAutoCF<CFAttributedStringRef>(CFAttributedStringCreate(nullptr, (CFStringRef)sc_frameText, dict));
+        auto framesetter = woc::MakeAutoCF<CTFramesetterRef>(CTFramesetterCreateWithAttributedString(attrString));
+        auto path = woc::MakeAutoCF<CGMutablePathRef>(CGPathCreateMutable());
+        CGPathAddRect(path, nullptr, CGRect{ CGPointZero, sc_defaultSize });
+        m_frame = woc::MakeAutoCF<CTFrameRef>(CTFramesetterCreateFrame(framesetter, CFRange{ 0, 0 }, path, nullptr));
+    }
+
+protected:
+    woc::AutoCF<CTFrameRef> m_frame;
+};
+
+class CTFrameDrawSingle : public CTFrameDrawBase {
+public:
+    inline void Run() {
+        CTFrameDraw(m_frame, m_context);
+    }
+
+    size_t GetRunCount() const {
+        return 100;
+    }
+};
+
+class CTFrameDrawGroup : public CTFrameDrawBase {
```

Comment:
DrawGroup tests:
 - should probably abstract out the PreRun PushBeginDraw/PostRun PopEndDraw to something common
 - I'm kind of uncomfortable with the n=1 runs for the group tests. I realize it's 100 frames, and that these tests take a while, but EndDraw only happens once, meaning that some small temporary hiccup on the machine we run this on could lead to us overinterpreting noise. n=5 maybe?